mono/mini/mini-trampolines.c

```diff
@@ -822,16 +813,21 @@ common_call_trampoline (mgreg_t *regs, guint8 *code, MonoMethod *m, MonoVTable *
 gpointer
 mono_magic_trampoline (mgreg_t *regs, guint8 *code, gpointer arg, guint8* tramp)
 {
-	MonoError error;
 	gpointer res;
 
+	MONO_PREPARE_RESET_BLOCKING_UNBALANCED;
+
+	MonoError error;
+
 	trampoline_calls ++;
 
 	res = common_call_trampoline (regs, code, (MonoMethod *)arg, NULL, NULL, &error);
-	if (!mono_error_ok (&error)) {
-		mono_error_set_pending_exception (&error);
-		return NULL;
-	}
+	mono_error_raise_exception (&error);
```

Comment:
I will do as @lambdageek advises
```
if (mono_error_set_pending_exception (&error))
  goto done;
```

On Mon, 9 May 2016, 19:14 Aleksey Kliger (λgeek), <notifications@github.com>
wrote:

> In mono/mini/mini-trampolines.c
> <https://github.com/mono/mono/pull/2969#discussion_r62589071>:
>
> >  	trampoline_calls ++;
> >
> >  	res = common_call_trampoline (regs, code, (MonoMethod *)arg, NULL, NULL, &error);
> > -	if (!mono_error_ok (&error)) {
> > -		mono_error_set_pending_exception (&error);
> > -		return NULL;
> > -	}
> > +	mono_error_raise_exception (&error);
>
> I'm pretty ambivalent about raise vs set_pending in this case: mostly we
> need to avoid raising in functions that are deep in the call tree from an
> interrupt check so that we can free resources (eventually local handles,
> for example). In this case the trampoline will start unwinding either way,
> so there's no hidden control flow.
>
> As a note, you can do:
>
> if (mono_error_set_pending_exception (&error))
>   goto done;
>
> which is slightly shorter.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mono/mono/pull/2969/files/7199756328be2f980bdbb987bbb43341dd42c455..e3573b59d1bcf2da1ac32cd3bcf4ee73e9752cd1#r62589071>
>